### ISSUE TITLE
Node GetAccessor did not pass test 'isTypeElement'.

### ISSUE THREAD

**Issue by lanwin** [microsoft/TypeScript#49211]
# Bug Report

### 🔎 Search Terms

"Node GetAccessor did not pass test 'isTypeElement'."

### 🕗 Version & Regression Information

While working on our web UI with Create React App this problem appears a lot in the last times. When I delete `node_modules\.cache\tsconfig.tsbuildinfo` and reastart Create React App it starts to work again, this is why I think its more a TS bug then related to Create React App. 

I currently have no idea what currently is triggering the bug. When I delete the cache it works for a while and then appears right out of nowhere again. 

CRA displays the following (development mode): 

> webpack compiled successfully
> Files successfully emitted, waiting for typecheck results...
> Issues checking in progress...
> Error: Debug Failure. False expression: Unexpected node.
> Verbose Debug Information: Node GetAccessor did not pass test 'isTypeElement'.
> Error: Debug Failure. False expression: Unexpected node.
> Verbose Debug Information: Node GetAccessor did not pass test 'isTypeElement'.
>     at visitNodes (C:\Dev\Repo\node_modules\typescript\lib\typescript.js:87664:39)
>     at Object.visitEachChild (C:\Dev\Repo\node_modules\typescript\lib\typescript.js:87884:60)
>     at visitDeclarationSubtree (C:\Dev\Repo\node_modules\typescript\lib\typescript.js:107162:31)
>     at Object.visitNode (C:\Dev\Repo\node_modules\typescript\lib\typescript.js:87593:23)
>     at transformTopLevelDeclaration (C:\Dev\Repo\node_modules\typescript\lib\typescript.js:107271:179)
>     at visitDeclarationStatements (C:\Dev\Repo\node_modules\typescript\lib\typescript.js:107227:26)
>     at Object.visitNodes (C:\Dev\Repo\node_modules\typescript\lib\typescript.js:87646:48)
>     at transformRoot (C:\Dev\Repo\node_modules\typescript\lib\typescript.js:106591:37)
>     at transformation (C:\Dev\Repo\node_modules\typescript\lib\typescript.js:107888:24)
>     at transformRoot (C:\Dev\Repo\node_modules\typescript\lib\typescript.js:107915:82)

Typescript: 4.6.4
react-scripts: 5.0.1

I did some test to fix it. Upgrade TS, remove node_modules, remove lockfile and reinstall. Nothing helped.

Our codebase is large and got a lot changes and since this problem can not reproduced, I need help in collection more information to narrow this issue to concrete problem.

**Comment by andrewbranch** [comment]
If you can give us a repo with reproduction instructions, we can investigate.

**Comment by lanwin** [comment]
> If you can give us a repo with reproduction instructions, we can investigate.

Thank you for your answer. The problem here is that this is a corporate private repostiory witch I can not share public.

**Comment by m0ksem** [comment]
> @m0ksem I see that ultimately runs a tool called `vue-tsc`. Can you reproduce the issue with `tsc`?

Same
```bash
(develop)~/projects/vuestic-ui/packages/ui$ yarn tsc src/main.ts --declaration --emitDeclarationOnly
```

```
vuestic-ui/node_modules/typescript/lib/tsc.js:73594
                        void ts.Debug.assertNode(visited, test);
                                      ^
Error: Debug Failure. False expression: Unexpected node.
Verbose Debug Information: Node 171 did not pass test 'isTypeElement'.
```

**Comment by andrewbranch** [comment]
@m0ksem thanks! Are you blocked by this or have you found a workaround?

**Comment by m0ksem** [comment]
> @m0ksem thanks! Are you blocked by this or have you found a workaround?

Moving to `4.6.3` from `4.6.4` helps!

**Comment by andrewbranch** [comment]
@lanwin filed the report against 4.6.4, so I guess we’re back at square one. @lanwin, for kicks do you want to try the latest TypeScript, 4.7.4?

**Comment by lanwin** [comment]
It seems that the lastest versions did not have this bug anymore. I did not encountered again lately.
